lib/ansible/plugins/inventory/__init__.py

```diff
@@ -428,6 +428,10 @@ def _add_host_to_keyed_groups(self, keys, variables, host, strict=False, fetch_h
                         prefix = keyed.get('prefix', '')
                         sep = keyed.get('separator', '_')
                         raw_parent_name = keyed.get('parent_group', None)
+                        default_value_name = keyed.get('default_value', None)
```

Comment:
I think this should apply to any host var value used for group names. Empty string values in lists/strings ignore this option. For example:
```
# inv1.yml
all:
  hosts:
    host1:
      dict_var: {"key": ""}
      list_var: [""]
      str_var: ""
```

```
# constructed.yml
plugin: constructed
keyed_groups:
  - key: dict_var
    default_value: value
  - key: list_var
    default_value: ignored
  - key: str_var
    default_value: ignored
```

```
$ ansible-inventory -i inv1.yml -i constructed.yml --graph
...
@all:
  |--@_:
  |  |--host1
  |--@_key_value:
  |  |--host1
```